README.md

```diff
@@ -262,77 +201,220 @@ builder.toValue();
+// {"op":"and","content":[{"op":"in","content":{"fieldName":"name","value":["Jim","Bob"]}},{"op":"gt","content":{"fieldName":"age","value":20}},{"op":"lt","content":{"fieldName":"age","value":50}}]}
+
+// Remove all filters on 'age'
+builder.removeFilter('age');
+// {"op":"in","content":{"fieldName":"name","value":["Jim","Bob"]}}
+```
+
+_Example removing some values from an array filter:_
+```ts
+const builder = SQONBuilder.in('name', ['Jim', 'Bob', 'May']);
+builder.removeFilter('name', FilterKeys.In, ['Jim', 'Bob', 'Sue']);
+// {"op":"in","content":{"fieldName":"name","value":["May"]}}
+```
+
+#### Set Filter
+`.setFilter(fieldName: string, op: FilterKey, value: FilterValueMap[FilterKey]) => SQONBuilder`
+
+Add a specific filter to the content of the top level operator, or replace a matching filter (same `op` and `fieldName`) with the new value specified.
+
+If the current SQON is just a filter, then the existing filter and this new filter will be combined with an `and` operator.
+
+Example:
+```ts
+const builder = SQONBuilder.gt('age', 50).in('name', ['Jim', 'Bob']);
+builder.setFilter('name', FilterKeys.In, ['Jim', 'Sue']);
+// {"op":"and","content":[{"op":"gt","content":{"fieldName":"age","value":50}},{"op":"in","content":{"fieldName":"name","value":["Jim","Sue"]}}]}
+```
+
+### Reduce SQON
+`reduceSQON(sqon: SQON) => SQON`
+
+The `reduceSQON` function is used internally by the SQON builder to reduce the complexity of a SQON by removing redundant operators and collecting similar filters. For example, if there is an `and` combination wrapping a single filter, the `and` operator can be removed and replaced by the single filter.
+
+The reducer function is exported so you are able to run the reduction algorithm on SQONs independent of the SQONBuilder.
+
+```ts
+import { reduceSQON } from '@overture-stack/sqon-builder`;
+
+const sqon: SQON = { op: 'and', 'content': [ { op: 'lt', content: { fieldName: 'age', value: 100 } } ] };
+
+const reduced = reduceSQON(sqon);
+// { op: 'lt', content: { fieldName: 'age', value: 100 } }
+```
+
+### Check Matching Filter
+
+`checkMatchingFilter(a: FilterOperator, b: FilterOperator) => boolean`
+
+The `checkMatchingFilter` function will compare two `FilterOperators` and return is they match in all properties. This ensures they have the same filter operation (`op` value), same `fieldName` and matching `value`. Note that for array values, the match is performed independent of order and after removing duplicates - it is a match on the logical content not on the exact array.
+
+```ts
+import { checkMatchingFilter } from '@overture-stack/sqon-builder`;
+const filterA = { op: FilterKeys.In, content: { fieldName: 'name', value: ['Jim', 'Bob', 'May'] } };
+const filterB = {	op: FilterKeys.In, content: { fieldName: 'name', value: ['May', 'Jim', 'Bob'] } };
+
+const matchResult = checkMatchingFilter(filterA, filterB); // true
+```
+
 
 ## Types and SQON Validation
```

Comment:
it occurs to me all this conceptual info can be moved before the methods, etc.

Reply:
Definitely needs a dedicated editing pass over the whole README.